src/spikeinterface/sorters/runsorter.py

```diff
@@ -500,58 +549,69 @@ def run_sorter_container(
+        need_si_install = "ModuleNotFoundError" in res_output
 
     if need_si_install:
-        if "dev" in si_version:
+        # update pip in container
+        cmd = f"pip install --user --upgrade pip"
+        res_output = container_client.run_command(cmd)
+
+        if installation_mode == "pypi":
             if verbose:
-                print(f"Installing spikeinterface from sources in {container_image}")
+                print(f"Installing spikeinterface=={si_version} with pypi in {container_image}")
 
-            # TODO later check output
-            if install_si_from_source:
-                si_source = "local machine"
-                # install in local copy of host SI folder in sources/spikeinterface to avoid permission errors
-                cmd = f"mkdir {si_source_folder}"
-                res_output = container_client.run_command(cmd)
-                cmd = f"cp -r {si_dev_path_unix} {si_source_folder}"
+            cmd = f"pip install --user --upgrade --no-input --no-build-isolation spikeinterface[full]=={si_version}"
+            res_output = container_client.run_command(cmd)
+        elif installation_mode == "github":
+            if verbose:
+                print(f"Installing spikeinterface from github sources in {container_image}")
+
+            if DEV_MODE:
+                # not released yet use main branch
+                cmd = "pip install --user --upgrade --no-input --no-build-isolation git+https://github.com/SpikeInterface/spikeinterface.git@main#egg=spikeinterface[full]"
                 res_output = container_client.run_command(cmd)
-                cmd = f"pip install --user {si_source_folder}/spikeinterface[full]"
             else:
-                si_source = "remote repository"
-                cmd = "pip install --user --upgrade --no-input git+https://github.com/SpikeInterface/spikeinterface.git#egg=spikeinterface[full]"
+                # already released and has a tag 
+                si_version = '0.99.1'
+                cmd = f"pip install --user --upgrade --no-input --no-build-isolation https://github.com/SpikeInterface/spikeinterface/archive/{si_version}.tar.gz#egg=spikeinterface[full]"
+                res_output = container_client.run_command(cmd)
+
+        elif host_folder_source is not None:
+            # this is "dev" + "folder"
             if verbose:
-                print(f"Installing dev spikeinterface from {si_source}")
+                print(f"Installing spikeinterface from local sources in {container_image}")
+
+            # copy spikeinterface sources to avoid problems between host and containers
+            cmd = f"mkdir {container_folder_source}"
             res_output = container_client.run_command(cmd)
-            cmd = "pip install --user --upgrade --no-input https://github.com/NeuralEnsemble/python-neo/archive/master.zip"
+            cmd = f"cp -r {container_folder_source_ro} {container_folder_source}"
```

Comment:
Are you sure you don't need a `cmd = f"pip install --user {container_folder_source}/spikeinterface[full]"` here?